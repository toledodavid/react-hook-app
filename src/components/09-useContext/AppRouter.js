import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route
} from 'react-router-dom';

import AboutScreen from './AboutScreen';
import HomeScreen from './HomeScreen';
import LoginScreen from './LoginScreen';



const AppRouter = () => {
  return(
    <Router>
      <div>

        <Switch>
          <Route exact path="/" component={HomeScreen} />
          <Route exact path="/login" component={LoginScreen} />
          <Route exact path="/about" component={AboutScreen} />
        </Switch>
      </div>
    </Router>
  );
}

export default AppRouter;