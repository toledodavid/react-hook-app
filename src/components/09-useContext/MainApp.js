import React from 'react';
import AppRouter from './AppRouter';
import { UserContext } from './UserContext';


const MainApp = () => {

  const user = {
    id: 1234,
    name: 'David',
    email: 'david@ucol.mx'
  }

  return(
    <UserContext.Provider value={user}>
      <AppRouter />
    </UserContext.Provider>
  );
}


export default MainApp;