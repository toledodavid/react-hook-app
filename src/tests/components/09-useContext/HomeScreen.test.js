import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import HomeScreen from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';


describe('Tests for <HomeScreen /> component', () => {

  const user = {
    name: 'David',
    email: 'david@gmail.com'
  };
  
  const wrapper = mount(
    <UserContext.Provider value={{user}}>
      <HomeScreen />
    </UserContext.Provider>
  );

  test('It should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
