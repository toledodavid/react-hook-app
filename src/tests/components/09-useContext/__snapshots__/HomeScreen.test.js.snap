// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <HomeScreen /> component It should render correctly 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "David",
   "email": "david@gmail.com"
}
  </pre>
</div>
`;
